### DISCUSSION TITLE
async postConstruct initialize in GLSPDiagramManager

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#1068] [Q&A]
Hi,
    I am migrating my theia application to 1.39.0, my glsp extensions are still 1.0.0. theia 1.39.0 use inversify 6.0.1, the problem is in GLSPDiagramManager
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/55d5bf10-4184-46b4-90c1-c123e4cd99a9)
the initialize method still return promise<void>
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/1dc7a1c0-8627-4362-bca1-b4bb9d2c712e)
 because now it is not fit to migrate to glsp 1.1.0.  Is there any method that can help to solve this problem in glsp version 1.0.0?

**Comment by tortmayr** [comment]
Hi @wss29,
GLSP 1.0.0 does only support Theia versions < 1.39.0. In theory you could rebind the `GLSPDiagramManager` to  a custom class and handle the postConstruct synchronously. However, since we have not tested this  you might run into additional issues when using GLSP 1.0.0 in Theia 1.39.
